Fix logout redirect to home

rkdavidson opened the pull request:
# About 
Closes #122

Moves the redirect to `/` to happen on logout click, instead of trying to do it all fancy-like up in the state provider.
I think this is a simpler, safer solution.

Also fixes a PropType warning on `CardGrid`.

# Testing
Log in and go to a page like **Discover**. Log out. Verify you're sent to home.
Navigate the site while unauthenticated, and sit on pages for ~5-10s to make sure you're not redirected randomly.

asleepingpanda reviewed (dismissed):
Looks great! Thanks @rkdavidson

rkdavidson commented:
Updated this PR to revert the log out redirect to home altogether.
Individual pages decide how to handle unauthenticated user access.

I updated **Connect** to redirect unauthenticated users to home, which I _think_ was the source problem in the first place 🤞 . **GroupSingle** will redirect to `/groups` for unauthenticated users, too.
![prevent-unauthenticated-connect](https://user-images.githubusercontent.com/1812955/116458513-bd23c000-a832-11eb-9502-1dec2a81f721.gif)
